app/metrics/numofrequests.go

```diff
@@ -0,0 +1,118 @@
+package metrics
+
+import (
+	"github.com/gorilla/websocket"
+	"github.com/koding/websocketproxy"
+	"github.com/prometheus/client_golang/prometheus"
+	"github.com/sirupsen/logrus"
+	"github.com/v3io/sidecar-proxy/app/utils"
+	"net/http"
+	"net/http/httputil"
+	"net/url"
+)
+
+var (
+	WebsocketUpgrader = utils.ExtendedWebSocket{
+		WebsocketUpgrader: websocket.Upgrader{
+			ReadBufferSize:  1024,
+			WriteBufferSize: 1024,
+		}}
+)
+
+type NumOfRequestsMetricsHandler struct {
+	logger         *logrus.Logger
+	forwardAddress string
+	listenAddress  string
+	namespace      string
+	serviceName    string
+	instanceName   string
+	metricName     MetricName
+	metric         *prometheus.CounterVec
+}
+
+func NewNumOfRequstsMetricsHandler(logger *logrus.Logger,
+	forwardAddress string,
+	listenAddress string,
+	namespace string,
+	serviceName string,
+	instanceName string) (*NumOfRequestsMetricsHandler, error) {
+	return &NumOfRequestsMetricsHandler{
+		logger:         logger,
+		forwardAddress: forwardAddress,
+		listenAddress:  listenAddress,
+		namespace:      namespace,
+		serviceName:    serviceName,
+		instanceName:   instanceName,
+		metricName:     NumOfRequestsMetricName,
+	}, nil
+}
+
+func (n *NumOfRequestsMetricsHandler) RegisterMetric() error {
+	requestsCounter := prometheus.NewCounterVec(prometheus.CounterOpts{
+		Name: string(n.metricName),
+		Help: "Total number of requests forwarded.",
+	}, []string{"namespace", "service_name", "instance_name"})
+
+	if err := prometheus.Register(requestsCounter); err != nil {
+		n.logger.WithError(err).WithField("metricName", string(n.metricName)).Error("Failed to register metric")
+		return err
+	}
+
+	n.logger.WithField("metricName", string(n.metricName)).Info("Metric registered successfully")
+	n.metric = requestsCounter
+
+	return nil
+}
+
+func (n *NumOfRequestsMetricsHandler) CollectData() {
```

Comment:
Why `CollectData`? This registers a handle, doesn't actively collect anything upon invocation

Reply:
Yeah, I agree - this interface IMO should be named something like `Start()` or something that "feels" async